app/src/main/java/com/stevesoltys/seedvault/ui/storage/StorageRootResolver.kt

```diff
@@ -34,6 +36,21 @@ internal object StorageRootResolver {
                     if (root != null) roots.add(root)
                 }
             }
+            // add special system user roots for USB devices
+            val c = context.getSystemContext {
+                authority == AUTHORITY_STORAGE && UserHandle.myUserId() != UserHandle.USER_SYSTEM
+            }
+            // only proceed if we really got a different [Context], e.g. had permission for it
+            if (context !== c) {
```

Comment:
I am not a 100% sure that this will work as expected. Mainly did it like this, so we don't run into a situation where the permission isn't granted and it uses the same context creating duplicate roots.

Reply:
Seems to work fine.